### PR TITLE
[litmus-portal] refactored, added support for different service types and ingress

### LINKED ISSUE #79: Service type hardcoded in litmus-portal chart

### LINKED ISSUE #80: Add ingress for litmus-portal chart

### PR DESCRIPTION
**Pull request by Jasstkn**
Signed-off-by: Jasstkn <jasssstkn@yahoo.com>
#### What this PR does / why we need it:
- litmus-portal refactored
- service type for each service is now configurable
- clean up structure of values, names
- added range over variables to leave deployment templates clean
- added ingress for frontend-svc

#### Which issue this PR fixes
  - fixes #79 
  - fixes #80 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] DCO signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by Jasstkn** [comment]
Hi! It's a draft for refactoring of litmus-portal changes. Please, check PR description.
Please, advice, if I need to test upgrade from previous version to this one. Because it's possible that it's not possible due to a lot of changes in the immutable objects..

**Comment by ksatchit** [comment]
I would like to get in @rajdas98 & @gdsoumya 's thoughts here.

**Comment by gdsoumya** [comment]
The changes look good to me, it would be good if you can post the helm install success logs and also show the pods that come up when the helm chart is installed @Jasstkn .

File: charts/litmus-portal/values.yaml [hunk 1]
```diff
@@ -27,26 +32,40 @@ portalFrontend:
...
+        pullPolicy: "Always"
+      containerPort: 8080
       env:
-        subscriberImage: "litmuschaos/litmusportal-subscriber:1.13.0"
-        argoServerImage: "argoproj/argocli:v2.9.3"
-        argoWorkflowControllerImage: "argoproj/workflow-controller:v2.9.3"
-        argoWorkflowExecutorImage: "argoproj/argoexec:v2.9.3"
-        litmusOperatorImage: "litmuschaos/chaos-operator:1.13.0"
-        litmusRunnerImage: "litmuschaos/chaos-runner:1.13.0"
+        SUBSCRIBER_IMAGE: "litmuschaos/litmusportal-subscriber:1.13.0"
```
Comment (rajdas98) [inline charts/litmus-portal/values.yaml:63]: it's better not to change the env syntax, we are using this format inside out code.

Comment (rajdas98) [reply charts/litmus-portal/values.yaml:63]: same for others

File: charts/litmus-portal/values.yaml [hunk 2]
```diff
@@ -8,14 +8,19 @@ adminConfig:
   DBUSER: "admin"
   DBPASSWORD: "1234"
   JWTSecret: "litmus-portal@123"
-  DataBaseServer: "mongodb://mongo-service:27017"
+  DataBaseServer: "mongodb://litmus-portal-mongo:27017"
```
Comment (rajdas98) [inline charts/litmus-portal/values.yaml:11]: For this, you need to change the mongodb service name also

**Review by rajdas98 — CHANGES_REQUESTED** [review]

File: charts/litmus-portal/values.yaml (hunk 1 above)
Comment (gdsoumya) [reply charts/litmus-portal/values.yaml:63]: This change is good @rajdas98 , the env vars we use in the actual server follow the caps format. The changes @Jasstkn has introduced will iterate over these env vars and put them directly into the server deployment as it is, so it should be good.

File: charts/litmus-portal/values.yaml (hunk 2 above)
Comment (gdsoumya) [reply charts/litmus-portal/values.yaml:11]: I think she has updated the db service too https://github.com/Jasstkn/litmus-helm/blob/a375d30bb113e38e41491819cea9d5cc97274961/charts/litmus-portal/templates/mongo-svc.yaml#L4

File: charts/litmus-portal/values.yaml (hunk 1 above)
Comment (Jasstkn) [reply charts/litmus-portal/values.yaml:63]: Yep, it allows to keep deployment manifest clean and transparent

File: charts/litmus-portal/values.yaml (hunk 2 above)
Comment (Jasstkn) [reply charts/litmus-portal/values.yaml:11]: I think that we can do even better, I'll make a commit.

Comment (Jasstkn) [reply charts/litmus-portal/values.yaml]: Done, re-check please :)

**Comment by Jasstkn** [comment]
> The changes look good to me, it would be good if you can post the helm install success logs and also show the pods that come up when the helm chart is installed @Jasstkn .

```
helm install litmus-portal .
NAME: litmus-portal
LAST DEPLOYED: Wed Mar  3 23:13:37 2021
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing litmus-portal 😀

Your release is named litmus-portal and it's installed to namespace: default.

Visit https://docs.litmuschaos.io/docs/getstarted/ to find more info.


kgp -w
NAME                                     READY   STATUS    RESTARTS   AGE
litmus-portal-frontend-6694f8849-lmbll   1/1     Running   0          2m9s
litmus-portal-mongo-756688bb57-42z4m     1/1     Running   0          2m9s
litmus-portal-server-578bc74b86-f8f8n    2/2     Running   0          35s
```

**Review by ksatchit — APPROVED** [review]

**Review by rajdas98 — APPROVED** [review]

**Pull request merged**